src/ApplicationInsights.Kubernetes/TelemetryInitializers/KubernetesTelemetryInitializer.cs

```diff
@@ -16,15 +18,19 @@ internal class KubernetesTelemetryInitializer : ITelemetryInitializer
 {
     private static readonly ApplicationInsightsKubernetesDiagnosticSource _logger = ApplicationInsightsKubernetesDiagnosticSource.Instance;
     private readonly SDKVersionUtils _sdkVersionUtils;
+    private readonly AppInsightsForKubernetesOptions _options;
+
     internal IK8sEnvironmentHolder K8SEnvironmentHolder { get; }
     internal ITelemetryKeyCache TelemetryKeyCache { get; }
 
     public KubernetesTelemetryInitializer(
         IK8sEnvironmentHolder k8sEnvironmentHolder,
         SDKVersionUtils sdkVersionUtils,
-        ITelemetryKeyCache telemetryKeyCache)
+        ITelemetryKeyCache telemetryKeyCache,
+        IOptions<AppInsightsForKubernetesOptions> options)
```

Comment:
It is an internal class so it does not matter that much, but if an uninitialized instance of options is valid, you could accept a null value and internally treat it the same as uninitialized instance.

Reply:
@karolz-ms, I don't think I understand what you are saying. Do you mind elaborate a bit? Like give me an example? Are you saying I could have accepted options to be null, instead of throwing?

Reply:
Let's have a coffee sometime and I will explain  😆